### PR TITLE
feat: add template module

### PR DESCRIPTION
**Pull request by luisdavim**
As discussed in https://github.com/risor-io/risor/discussions/68

### CODE REVIEW

File: modules/template/jsonpath.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+		return nil, fmt.Errorf("error executing jsonpath: %w", err)
+	}
+
+	return strings.Trim(buf.String(), "\n"), nil
+}
+
+func envMap() map[string]string {
+	envMap := make(map[string]string)
+
+	for _, v := range os.Environ() {
```
Comment (myzie) [inline modules/template/jsonpath.go:68]: Will you update this to use the Risor `OS` interface? This lets the env come from a dynamic/configurable source.

You can call this:
https://github.com/risor-io/risor/blob/main/os/os.go#L95

Which means you'll have to pass in a context.

File: modules/template/template.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+	if err := Render(buf, template, data.Interface()); err != nil {
+		return object.NewError(err)
+	}
+
+	return object.NewString(buf.String())
+}
+
+func Builtins() map[string]object.Object {
+	return map[string]object.Object{
+		"template": object.NewBuiltin("template", Template),
```
Comment (myzie) [inline modules/template/template.go:38]: The `template` function seems to work great.

As someone that just gave it a try without much review first, at first I didn't expect to have to use `.Values` as the prefix to use the input data. To what degree do you think of this @luisdavim as a _Helm_ template equivalent vs. a Go template equivalent?

I wonder if both should be offered in Risor. If both were offered, would it make sense for the Helm version to be known as the global builtin `template` or would that most logically correspond to Go's `text/template` package?

I'm not sure of the "right" answer yet... this is just something I'm thinking through real quick.

File: modules/template/jsonpath.go (hunk 1 above)
Comment (luisdavim) [reply modules/template/jsonpath.go:68]: sure, I'll look into it.

File: modules/template/template.go (hunk 2 above)
Comment (luisdavim) [reply modules/template/template.go:38]: The main reason ti place the passed values under `Values` (or any other field) is to keep them separate from any builtin sets of values we might have and avoid collisions, in this case we have the `Env` builtin set of values and in the future we might want to add more...

I'm ok with renaming or even removing both the `.Values` and `.Env`....

File: modules/template/jsonpath.go (hunk 1 above)
Comment (luisdavim) [reply modules/template/jsonpath.go]: updated.

File: modules/template/template.go (hunk 2 above)
Comment (myzie) [reply modules/template/template.go:38]: I think having one builtin that operates exactly in this way makes a ton of sense and accessing `.Env` values will be super useful in Risor, I expect.

But we may want a separate builtin that is a vanilla Go text/template without .Values and .Env. 

So if we have both options, I'm not yet sure what the best names for each would be. And whether one or both should be offered in a `template` module. If we did a module we could still have the module object itself be callable, which is something I did in one or two modules already.

Comment (luisdavim) [reply modules/template/template.go:38]: Yeah but if we do have this option with "batteries included" (out of the box values and functions) is it really need to also provide a more barebones option that is closer to what Go text/template provides? Do you have any use cases for that in mind?

Comment (myzie) [reply modules/template/template.go:38]: The `.Values` approach only has the downsides that 1) it's not intuitively guessable and 2) it's slightly more typing when you're just feeding in a simple map.

If someone comes to Risor expecting the `template` builtin to correspond to Go's `text/template` package, which would be reasonable in general, they would have no idea that they have to use `.Values` in their template.

I'm totally fine with having this version of the `template` builtin to be the one and only version for now. But I'm just wanting to allow for the other version to be added later without creating naming confusion.

Comment (luisdavim) [reply modules/template/template.go:38]: OK, what about some approach like what we have with the http module, a built-in that is the current implementation, maybe named render_template or just render and a module that allows initiate a template, set the delimiters and maybe some other options and then render a template, this one would be closer to the default golang template and have no default values.

BTW, I agree the .Values is not intuitive or easily discoverable nut that also a matter of documentation.

Comment (myzie) [reply modules/template/template.go:38]: Yeah, just using the name “render” - for what you have here - would be enough for now. That’ll let us reserve the name “template” for the module, whenever we do that.

If you just do that simple rename we can get this PR merged.

Having this builtin named as a verb makes some sense too, another plus.

Comment (luisdavim) [reply modules/template/template.go]: @myzie I've updated the PR, renamed the original builtin to `render` and added a `template` module, we can add things like `parse_files` to it latter but I think this is a good starting point, please let me know what you think.

**Comment by myzie** [comment]
This is great. Thanks for working with me on those little adjustments.

You made a great choice with having the data be the first argument to the `render` function. Being able to use a pipe statement to feed the input data into the rendering feels superb.

The examples you've been adding are really helpful for reviewing as well.

**Pull request merged**